### PR TITLE
Show more details when token refresh fails

### PR DESCRIPTION
**Pull request by hadley**
@jennybc do you mind taking a careful look at this? I can't think of any way to test this so I think it needs a careful read through to make sure I haven't introduced any new bugs.

### CODE REVIEW

File: R/oauth-error.r [hunk 1]
```diff
@@ -14,12 +14,17 @@ oauth2.0_errors <- c(
...
-  if (status_code(response) %in% oauth2.0_error_codes) {
-    content <- content(response)
-    if (content$error %in% oauth2.0_errors) {
-      return(TRUE)
-    }
-  }
-  FALSE
+find_oauth2.0_error <- function(response) {
+  if (!status_code(response) %in% oauth2.0_error_codes)
+    return(NULL)
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/oauth-error.r]: I like bracketless for these very short early returns

[the remark this answers is not in the document]
Comment (hadley) [reply R/oauth-error.r]: Regardless, the current style is such a mismash, I don't think it's worth fixing.

**Comment by hadley** [comment]
It would be nice if google followed the spec here, but making this extensible is just something we'll need to carefully think through in httr2.

**Comment by jennybc** [comment]
> It would be nice if google followed the spec here...

This token refresh part is what really matters, right? Does it depart from the spec?

```
Auto-refreshing stale OAuth token.
Warning in refresh_oauth2.0(self$endpoint, self$app, self$credentials, self$params$user_params,  :
  Unable to refresh token: invalid_grant
Token has been expired or revoked.
```

The content of the `GET()` just reflects what we get when we send an invalid token. I hope I didn't confuse matters by including that.

**Comment by hadley** [comment]
Oh I see what you mean now.

**Pull request merged**